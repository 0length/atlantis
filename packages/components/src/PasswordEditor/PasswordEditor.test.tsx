import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { PasswordEditor } from ".";

afterEach(cleanup);

test("renders", () => {
  const tree = renderer
    .create(<PasswordEditor value="foo" onChange={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders with custom placeholder", () => {
  const tree = renderer
    .create(
      <PasswordEditor placeholder="Secrets" value="foo" onChange={jest.fn()} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders an error message when given an insecure value", () => {
  const tree = renderer
    .create(<PasswordEditor value="password" onChange={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("it should call the handler with a new secure value", () => {
  const changeHandler = jest.fn();
  const newPassword = "oh, look a password";

  const { getByLabelText } = render(
    <PasswordEditor value="" onChange={changeHandler} />,
  );

  fireEvent.change(getByLabelText("Password"), {
    target: { value: newPassword },
  });
  expect(changeHandler).toHaveBeenCalledWith([newPassword, true]);
});

test("it should call the handler with a new insecure value", () => {
  const changeHandler = jest.fn();
  const newPassword = "password";

  const { getByLabelText } = render(
    <PasswordEditor value="" onChange={changeHandler} />,
  );

  fireEvent.change(getByLabelText("Password"), {
    target: { value: newPassword },
  });
  expect(changeHandler).toHaveBeenCalledWith([newPassword, false]);
});

test("it should display an appropriate strength gauge", () => {
  const changeHandler = jest.fn();

  const { queryByLabelText } = render(
    <PasswordEditor value="passwordBuddy" onChange={changeHandler} />,
  );

  expect(queryByLabelText("Level 0")).not.toBeInstanceOf(HTMLElement);
  expect(queryByLabelText("Level 1")).toBeInstanceOf(HTMLElement);
  expect(queryByLabelText("Level 2")).not.toBeInstanceOf(HTMLElement);
  expect(queryByLabelText("Level 3")).not.toBeInstanceOf(HTMLElement);
  expect(queryByLabelText("Level 4")).not.toBeInstanceOf(HTMLElement);
});
