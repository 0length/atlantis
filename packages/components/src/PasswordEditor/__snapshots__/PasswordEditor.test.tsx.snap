// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="padded base"
>
  <div
    className="padded small"
  >
    <p
      className="base regular base red"
    >
      Add another word or two. Uncommon words are better.
    </p>
    
  </div>
  <div
    className="wrapper miniLabel"
    style={
      Object {
        "--formField-maxLength": undefined,
      }
    }
  >
    <label
      className="label"
      htmlFor="123e4567-e89b-12d3-a456-426655440001"
    >
      Password
    </label>
    <input
      className="formField"
      id="123e4567-e89b-12d3-a456-426655440001"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="password"
      value="foo"
    />
  </div>
  <div
    aria-label="Level 0"
    className="strengthGauge level0"
  />
</div>
`;

exports[`renders an error message when given an insecure value 1`] = `
<div
  className="padded base"
>
  <div
    className="padded small"
  >
    <p
      className="base regular base red"
    >
      Add another word or two. Uncommon words are better.
    </p>
    <p
      className="base regular base red"
    >
      This is a top-10 common password
    </p>
  </div>
  <div
    className="wrapper miniLabel"
    style={
      Object {
        "--formField-maxLength": undefined,
      }
    }
  >
    <label
      className="label"
      htmlFor="123e4567-e89b-12d3-a456-426655440003"
    >
      Password
    </label>
    <input
      className="formField"
      id="123e4567-e89b-12d3-a456-426655440003"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="password"
      value="password"
    />
  </div>
  <div
    aria-label="Level 0"
    className="strengthGauge level0"
  />
</div>
`;

exports[`renders with custom placeholder 1`] = `
<div
  className="padded base"
>
  <div
    className="padded small"
  >
    <p
      className="base regular base red"
    >
      Add another word or two. Uncommon words are better.
    </p>
    
  </div>
  <div
    className="wrapper miniLabel"
    style={
      Object {
        "--formField-maxLength": undefined,
      }
    }
  >
    <label
      className="label"
      htmlFor="123e4567-e89b-12d3-a456-426655440002"
    >
      Secrets
    </label>
    <input
      className="formField"
      id="123e4567-e89b-12d3-a456-426655440002"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="password"
      value="foo"
    />
  </div>
  <div
    aria-label="Level 0"
    className="strengthGauge level0"
  />
</div>
`;
